syntax = "proto3";

package Protocol;

message C2S_AccountLogin {
    int32 game_id = 1;
    string openid = 2;
}

message S2C_AccountLogin {
    int32 ret = 1;
    int64 role_id = 2;
}

message C2S_AccountCreate {
    int32 game_id = 1;
    string openid = 2;
}

message S2C_AccountCreate {
    int32 ret = 1;
    int64 role_id = 2;
}

message C2S_AccountLoad {
    int32 game_id = 1;
    int64 role_id = 2;
}

message S2C_AccountLoad {
    int32 ret = 1;
    int64 role_id = 2;
}

message C2S_AccountAskOpenid {
    int32 game_id = 1;
    string code = 2;
    string anonymous_code = 3;
}

message S2C_AccountAskOpenid {
    int32 ret = 1;
    string openid = 2;
}

message C2S_AccountUpdate {
    int32 game_id = 1;
    string role_id = 2;
    string name = 3;
    string avatarurl = 4;
    string country = 5;
    string city = 6;
}

message S2C_AccountUpdate {
    int32 ret = 1;
}